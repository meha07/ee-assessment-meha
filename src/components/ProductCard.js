import React from 'react';
// import Card from '@mui/material/Card';
// import CardContent from '@mui/material/CardContent';
// import CardMedia from '@mui/material/CardMedia';
// import Typography from '@mui/material/Typography';
// import { CardActionArea } from '@mui/material';

const ProductCard = ({ product }) => {
  
  return (
      <div className="product-card">
        <div className="product-image"><img src={product.image} alt={product.title} /></div>
        <div className="product-description">
          <h3>{product.title}</h3>
          <p>${product.price}</p>
        </div>
      </div>
  );
}

export default ProductCard;