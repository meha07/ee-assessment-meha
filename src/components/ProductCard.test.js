import React from 'react';
import { render } from '@testing-library/react';
import ProductCard from './ProductCard';
import '@testing-library/jest-dom'

const sampleProduct = {
    title: 'Acer SB220Q bi 21.5 inches Full HD (1920 x 1080) IPS Ultra-Thin',
    price: 599,
    image: 'Acer SB220Q bi 21.5 inches Full HD (1920 x 1080) IPS Ultra-Thin',
};

describe('ProductCard Component', () => {
  it('renders product title', () => {
    const { getByText } = render(<ProductCard product={sampleProduct} />);
    const titleElement = getByText('Acer SB220Q bi 21.5 inches Full HD (1920 x 1080) IPS Ultra-Thin');
    expect(titleElement).toBeInTheDocument();
  });

  it('renders product price', () => {
    const { getByText } = render(<ProductCard product={sampleProduct} />);
    const priceElement = getByText('$599');
    expect(priceElement).toBeInTheDocument();
  });

  it('renders product image', () => {
    const { getByAltText } = render(<ProductCard product={sampleProduct} />);
    const imageElement = getByAltText('Acer SB220Q bi 21.5 inches Full HD (1920 x 1080) IPS Ultra-Thin');
    expect(imageElement).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<ProductCard product={sampleProduct} />);
    expect(asFragment()).toMatchSnapshot();
  });
});