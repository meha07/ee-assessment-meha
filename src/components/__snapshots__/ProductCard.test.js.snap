// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="product-card"
  >
    <div
      class="product-image"
    >
      <img
        alt="Acer SB220Q bi 21.5 inches Full HD (1920 x 1080) IPS Ultra-Thin"
        src="Acer SB220Q bi 21.5 inches Full HD (1920 x 1080) IPS Ultra-Thin"
      />
    </div>
    <div
      class="product-description"
    >
      <h3>
        Acer SB220Q bi 21.5 inches Full HD (1920 x 1080) IPS Ultra-Thin
      </h3>
      <p>
        $599
      </p>
    </div>
  </div>
</DocumentFragment>
`;
